cmd/k8s-operator/ingress-for-pg.go

```diff
@@ -44,6 +44,8 @@ const (
 	VIPSvcOwnerRef = "tailscale.com/k8s-operator:owned-by:%s"
 	// FinalizerNamePG is the finalizer used by the IngressPGReconciler
 	FinalizerNamePG = "tailscale.com/ingress-pg-finalizer"
+
+	indexIngressProxyGroup = ".metadata.annotations.ingress-proxy-group"
```

Comment:
I could put this change into a separate PR- this was mostly me noticing that if the Ingress is created before the ProxyGroup, it does not get reconciled when the ProxyGroup is created- so I just copy-pasted the same indexing approach we use to ensure that egress services are reconciled on egress ProxyGroup